### PR TITLE
Enhancement/#16 version check

### PR DESCRIPTION
**Pull request by MahdiZareie**
This PR mainly is about Version checking( #16 ) but also contains other small bugfixes:
 *  feedback message on `fandogh image versions` command #22 
 *  handling 400 response in a different way
 * Redesigning config module
 * Adding some utility to format_text with colors in terminals

### CODE REVIEW

File: fandogh_cli/__init__.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+VERSION = "0.2.3"
```
Comment (s-soroosh) [inline fandogh_cli/__init__.py:1]: why do we need these attributes?

File: fandogh_cli/fandogh_client/__init__.py [hunk 2]
```diff
@@ -77,6 +88,8 @@ def create_version(image_name, version, workspace_path):
 
 def list_versions(image_name):
     response = requests.get(base_images_url + '/' + image_name + '/versions')
+    if response.status_code == 404:
+        return ""
```
Comment (s-soroosh) [inline fandogh_cli/fandogh_client/__init__.py]: why empty string here?

File: fandogh_cli/version_check.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            return response.json()['releases'].keys()
+        raise RuntimeError("Unexpected response status while calling pypi")
+    except KeyError as missing_key:
+        raise RuntimeError("Unexpected response from pypi, missing key: {}".format(missing_key))
+    except Exception as exp:
+        raise ConnectionError("Unable to connect to pypi.org: {}".format(exp))
+
+
+def get_max_version():
+    all_versions = sorted([Version(v) for v in get_versions()], reverse=True)
```
Comment (s-soroosh) [inline fandogh_cli/version_check.py]: you don't need to iterated over versions to fine the latest.
info.version should be good.

File: fandogh_cli/version_check.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        if response.status_code == 200:
+            return response.json()['releases'].keys()
+        raise RuntimeError("Unexpected response status while calling pypi")
+    except KeyError as missing_key:
+        raise RuntimeError("Unexpected response from pypi, missing key: {}".format(missing_key))
+    except Exception as exp:
+        raise ConnectionError("Unable to connect to pypi.org: {}".format(exp))
+
+
+def get_max_version():
```
Comment (s-soroosh) [inline fandogh_cli/version_check.py]: get_latest_version?

File: fandogh_cli/__init__.py (hunk 1 above)
Comment (MahdiZareie) [reply fandogh_cli/__init__.py:1]: We need to have access to version number inside the package, I can't read it from `setup.py`

Comment (s-soroosh) [reply fandogh_cli/__init__.py:1]: Check fandogh —version

File: fandogh_cli/fandogh_client/__init__.py (hunk 2 above)
Comment (MahdiZareie) [reply fandogh_cli/fandogh_client/__init__.py]: Empty string signals the caller that there is no version available

File: fandogh_cli/version_check.py (hunk 3 above)
Comment (MahdiZareie) [reply fandogh_cli/version_check.py]: Damn :-/... thanks :+1:

**Comment by MahdiZareie** [comment]
@psycho-ir I tested this PR on py2 py3, it's good to go

File: fandogh_cli/fandogh_client/__init__.py (hunk 2 above)
Comment (s-soroosh) [reply fandogh_cli/fandogh_client/__init__.py]: No version? We return empty list if there is no version.
404 might mean image does not exist

**Review by s-soroosh — APPROVED** [review]

**Pull request merged**